syntax = "proto3";

package bookstore;

import "shared/order.proto";

service SuggestionsService {
    rpc InitGetSuggestions (SuggestionsRequest) returns (ErrorResponse);
    rpc GetSuggestions (ExecInfo) returns (ErrorResponse);
}

message SuggestionsRequest {
    ExecInfo info = 1;
    repeated Book booksInCart = 2;
}

message SuggestionsResponse {
    repeated Book booksSuggested = 1;
}

message Book {
    int64 bookId = 1;
    string title = 2;
    string author = 3;
}